### ISSUE TITLE
Follow psalm changes in patch version

### ISSUE THREAD

**Issue by adamturcsan** [Roave/you-are-using-it-wrong#114]
Change in Psalm which is used by this project:
https://github.com/vimeo/psalm/commit/205fdd197e914a5377053f9378a7112236a83b50#diff-58b56f2d1a81b22decaddcb6bce7b755662ebf12b8e17e20fd60dc5809787c99

I'm not familiar with Psalm's versioning policy and its public/internal APIs but the new version (4.6.3) is crashing with this project.

What is happening right now:
```
Fatal error: Uncaught TypeError: Argument 2 passed to Psalm\IssueBuffer::finish() must be an instance of Psalm\Internal\ExecutionEnvironment\BuildInfoCollector, bool given, called in /builds/jarkon/modules/reminder/vendor/roave/you-are-using-it-wrong/src/Hook.php on line 108 and defined in /builds/jarkon/modules/reminder/vendor/vimeo/psalm/src/Psalm/IssueBuffer.php:442
Stack trace:
#0 /builds/jarkon/modules/reminder/vendor/roave/you-are-using-it-wrong/src/Hook.php(108): Psalm\IssueBuffer::finish(Object(Psalm\Internal\Analyzer\ProjectAnalyzer), true, 1615890391.2501)
#1 /builds/jarkon/modules/reminder/vendor/roave/you-are-using-it-wrong/src/Hook.php(76): Roave\YouAreUsingItWrong\Hook::analyseProject(Object(Roave\YouAreUsingItWrong\Composer\Project))
#2 [internal function]: Roave\YouAreUsingItWrong\Hook::runTypeChecks(Object(Composer\Script\Event))
#3 phar:///builds/jarkon/modules/reminder/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php(174): call_user_func(Array, Object(Composer\Script\Event))
#4 phar:///bui in /builds/jarkon/modules/reminder/vendor/vimeo/psalm/src/Psalm/IssueBuffer.php on line 442
```

It may be a Psalm issue as it is a BC break in a patch version.

**Comment by Ocramius** [comment]
Yeah, seems to be blocked at upgrade: https://github.com/Roave/you-are-using-it-wrong/pull/113

Feel free to send a patch including #113 changes, so we can fix it here, but overall this should be reported as upstream BC break too.

**Comment by weirdan** [comment]
I'd suggest to explicitly list 4.6.3 as a conflicting version. I hope this is going to be fixed in 4.6.4 (by https://github.com/vimeo/psalm/pull/5410).

> Yeah, seems to be blocked at upgrade: #113

As far as I can tell upgrades are failing for a different reason. Psalm 4.5.0+ emits exit code 2 when it identifies issues. Specific exit code wasn't documented before that version, but tests in this repo assumed it would be 1. I've fixed issues I was able to identify in #115
